codex-rs/mcp-server/src/lib.rs

```diff
@@ -77,10 +82,26 @@ pub async fn run_main(codex_linux_sandbox_exe: Option<PathBuf>) -> IoResult<()>
         }
     });
 
+    // Parse CLI overrides once so we can reuse them for operations
+    // like `get_auth_status` that need access to the effective config.
+    let cli_kv_overrides = match cli_config_overrides.parse_overrides() {
+        Ok(v) => v,
+        Err(e) => {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                format!("error parsing -c overrides: {e}"),
+            ));
+        }
+    };
+
     // Task: process incoming messages.
     let processor_handle = tokio::spawn({
         let outgoing_message_sender = OutgoingMessageSender::new(outgoing_tx);
-        let mut processor = MessageProcessor::new(outgoing_message_sender, codex_linux_sandbox_exe);
+        let mut processor = MessageProcessor::new(
+            outgoing_message_sender,
+            codex_linux_sandbox_exe,
+            cli_kv_overrides,
```

Comment:
I think `cli_kv_overrides` is an awkward type to thread through. `MessageProcessor` should not have to know about `TomlValue`, for example. I would derive the `Config` from it as early as possible and pass that through instead.